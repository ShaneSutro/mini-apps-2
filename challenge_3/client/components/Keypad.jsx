import React from 'react';

const Keypad = ({ actions }) => (
  <div>
    <table>
      <tbody>
        <tr>
          <td onClick={actions.select}>1</td>
          <td onClick={actions.select}>2</td>
          <td onClick={actions.select}>3</td>
          <td onClick={actions.select}>4</td>
        </tr>
        <tr>
          <td onClick={actions.select}>5</td>
          <td onClick={actions.select}>6</td>
          <td onClick={actions.select}>7</td>
          <td onClick={actions.select}>8</td>
        </tr>
        <tr>
          <td onClick={actions.select}>9</td>
          <td onClick={actions.select}>10</td>
          <td> </td>
          <td onClick={actions.select}>0</td>
        </tr>
      </tbody>
    </table>
  </div>
);

export default Keypad;
