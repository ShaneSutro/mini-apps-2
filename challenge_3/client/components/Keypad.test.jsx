/* eslint-disable no-undef */
import React from 'react';
import { shallow } from 'enzyme';

import Keypad from './Keypad.jsx'

describe('Testing the scoring algorithms', () => {
  test('Should be set up correctly', () => {
    const keypad = shallow(<Keypad actions={{ select: () => { } }} />);
    expect(keypad).toMatchSnapshot();
  });

  test('Should be a 4 by 3 grid', () => {
    const keypad = shallow(<Keypad actions={{ select: () => { } }} />);
    expect(keypad.find('td').length).toBe(12);
  });

  test('Should call the actions.select function when clicked', () => {
    const select = jest.fn();
    const keypad = shallow(<Keypad actions={{ select }} />);
    keypad.find('tbody').childAt(0).childAt(0).simulate('click');
    expect(select).toHaveBeenCalled();
  });
});
