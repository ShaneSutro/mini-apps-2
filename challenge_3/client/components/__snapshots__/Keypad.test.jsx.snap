// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing the scoring algorithms Should be set up correctly 1`] = `
<div>
  <table
    id="keypad"
  >
    <tbody>
      <tr>
        <td
          onClick={[Function]}
        >
          1
        </td>
        <td
          onClick={[Function]}
        >
          2
        </td>
        <td
          onClick={[Function]}
        >
          3
        </td>
        <td
          id="four"
          onClick={[Function]}
        >
          4
        </td>
      </tr>
      <tr>
        <td
          onClick={[Function]}
        >
          5
        </td>
        <td
          onClick={[Function]}
        >
          6
        </td>
        <td
          onClick={[Function]}
        >
          7
        </td>
        <td
          onClick={[Function]}
        >
          8
        </td>
      </tr>
      <tr>
        <td
          onClick={[Function]}
        >
          9
        </td>
        <td
          onClick={[Function]}
        >
          10
        </td>
        <td>
           
        </td>
        <td
          onClick={[Function]}
        >
          0
        </td>
      </tr>
    </tbody>
  </table>
</div>
`;
